wordpress-mobile/WordPress-iOS
# UL&S: PasswordViewController: send magic link

Ref. #14009
Auth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/403/
Design review [approved](https://github.com/wordpress-mobile/WordPress-iOS/pull/14746) in a previous PR.

This PR:
- adds the ability to send a magic link on the password entry screen
- updates the button color in Unified to "hero blue" instead of .brand
- (doesn't have Tracks set up for the magic link event)

### To test
Use a WordPress.com account that can be connected to a social account (such as gmail / google).

1. `rake dependencies`
2. Build and run
3. Log In > Continue with Google
4. Log into Google and wait for the password challenge view to appear.
5. Notice the new button in the PasswordViewController.
6. Select the "Continue" button without entering a password. **Expected:** an error message appears.
7. Select "Get a login link by email". **Expected:** you are navigated to the "magic link sent!" screen.

### Screenshots

| Design | Screenshot |
| --- | --- |
| <img width="344" alt="Screen Shot 2020-08-24 at 8 12 28 AM" src="https://user-images.githubusercontent.com/1062444/91182398-bd6e2900-e6af-11ea-8447-206884d5bb01.png"> | <a href="https://user-images.githubusercontent.com/1062444/91183386-0672ad00-e6b1-11ea-82d1-13e50029122d.png"><img src="https://user-images.githubusercontent.com/1062444/91183386-0672ad00-e6b1-11ea-82d1-13e50029122d.png" width="350" /></a> |

| Password errors are in-line | Magic link errors are alerts |
| --- | --- |
| <a href="https://user-images.githubusercontent.com/1062444/91183426-15f1f600-e6b1-11ea-805d-6d8038164a7a.png"><img src="https://user-images.githubusercontent.com/1062444/91183426-15f1f600-e6b1-11ea-805d-6d8038164a7a.png" width="350" /></a> | <a href="https://user-images.githubusercontent.com/1062444/91183459-20ac8b00-e6b1-11ea-9667-f29e1337adbd.png"><img src="https://user-images.githubusercontent.com/1062444/91183459-20ac8b00-e6b1-11ea-9667-f29e1337adbd.png" width="350" /></a> |

PR submission checklist:
